Azure/sap-hana
# Add the subscription id and tenant id to the output.json file for use by Ansible

## Problem

As part of #338, the need to have the Subscription ID and Tenant ID available to the Ansible code for HA configuration for the Clustering STONITH Service Principal.

## Solution

This PR uses the `azurerm_client_config` data source in Terraform to add the required details to the `output.json` file.

## Test Instructions/Acceptance Criteria

:hand: The following testing was performed in [PR #22 in the Centiq fork](https://github.com/Centiq/sap-hana/pull/22).

1. Ensure your working directory is clean:\
   `util/terraform_v2.sh clean`
   `y`
1. Initialise the terraform modules:\
   `util/terraform_v2.sh init`
1. Update the resource group in the `rti_only` template (make sure this is personalised/not going to conflict with any other resources), e.g.:\
   `util/set_resource_group.sh "pc-tf-test-rg" "rti_only"`
1. Build the infrastructure:\
   `util/terraform_v2.sh apply rti_only`
1. Note the `id` and `tenantId` values for the AZ Account information:\
   `az account show`
1. Using the output from the above, SSH to the RTI instance:\
   `ssh azureadm@<IP_ADDRESS>`
1. Check the `output.json` file for the azure configuration:\
   `cat output.json | |python -m json.tool | grep -A3 azure_config`
   - [x] The `subscription_id` value matches the above noted `id` value
   - [x] The `tenant_id` value matches the above noted `tenantId` value

## Post-Testing

:hand: Make sure you delete the resource group created for this test after completion.
